### PR TITLE
#2019 - Bug fix MobileAppRegistry is a singleton

### PR DESCRIPTION
**Pull request by MackHalliday**
# Description

issue #2019 

- This PR updates how the `MobileAppRegistry` class is initialized. Move initialization to `app/__init__.py` to implement as a singleton. 
- Delete incorrect test for "Singleton" 
- Update MobileAppRegistry class in the health status check endpoint and in the `push_notification_helper`

## How Has This Been Tested?

[Ran regression suite - OK](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/11600511523)
- [Hit GET `/_status` and confirmed that new log `Initializing MobileAppRegistry` does not appear](https://vanotify.ddog-gov.com/logs?query=host%3Adev-notification-api-log-group&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=time%2Cdesc&viz=stream&from_ts=1730317530218&to_ts=1730318430218&live=false)

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

File: app/mobile_app/mobile_app_registry.py [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
-        if not cls._instance:
-            cls._instance = super().__new__(cls)
-        return cls._instance
-
-    def __init__(self):
+    def __init__(self, logger):
+        self.logger = logger
         self._registry = {}
+
+        logger.info('Intializing MobileAppRegistry')
```
Comment (MackHalliday) [inline app/mobile_app/mobile_app_registry.py:12]: @David This I included temporarily to test on DEV to test this was only initialized once.  I planned on removing this log.

Comment (kalbfled) [reply app/mobile_app/mobile_app_registry.py]: I would leave it.  It should only be called once.

File: tests/app/v2/notifications/test_push_broadcast_notifications.py [hunk 2]
```diff
@@ -185,11 +185,15 @@ def test_makes_call_to_vetext_client(
...
+        """
+        App SIDs should be loaded from the environment.  For testing, the is .local.env.
+        """
+
         service = sample_service(service_permissions=[PUSH_TYPE])
 
         post_send_push_broadcast_notification(client, sample_api_key(service), payload)
         vetext_client.send_push_notification.assert_called_once_with(
-            f'some_sid_for_{app}', payload['template_id'], payload['topic_sid'], True, personalisation
+            f'{app}_sid', payload['template_id'], payload['topic_sid'], True, personalisation
```
Comment (MackHalliday) [inline tests/app/v2/notifications/test_push_broadcast_notifications.py:196]: Corrected test.

File: app/mobile_app/mobile_app_registry.py [hunk 3]
```diff
@@ -1,35 +1,27 @@
 from typing import List
-from flask import current_app
-from .mobile_app_types import MobileAppType
+
 from .mobile_app import MobileApp
+from .mobile_app_types import MobileAppType
 
 
 class MobileAppRegistry:
-    _instance = None
```
Comment (MackHalliday) [inline app/mobile_app/mobile_app_registry.py:8]: Does not actually make class a Singleton in the scope of the entire application.

File: tests/app/mobile_app/test_mobile_app_registry.py [hunk 4]
```diff
@@ -1,20 +1,15 @@
...
+from app.mobile_app.mobile_app_registry import MobileAppRegistry
+from app.mobile_app.mobile_app_types import MobileAppType
 
-@pytest.fixture(autouse=True)
-def mock_logger(mocker):
-    app_context_mock = mocker.patch('app.mobile_app.mobile_app_registry.current_app')
-    return app_context_mock.logger
 
-
-def test_registry_is_singleton(
```
Comment (MackHalliday) [inline tests/app/mobile_app/test_mobile_app_registry.py:12]: Test not actually testing if MobileAppRegistry is a singleton.

File: app/__init__.py [hunk 5]
```diff
@@ -213,6 +217,9 @@ def create_app(application):
 
     jwt.init_app(application)
 
+    global mobile_app_registry
```
Comment (MackHalliday) [inline app/__init__.py:220]: Make `mobile_app_registry` accessible thru `import mobile_app_registry from app`

File: tests/app/v2/notifications/test_push_notifications.py [hunk 6]
```diff
@@ -267,18 +267,3 @@ def test_maps_bad_request_exception(
         assert response.status_code == 400
         resp_json = response.get_json()
         assert {'error': 'BadRequestError', 'message': exception.message} in resp_json['errors']
-
-    @pytest.mark.disable_autouse
```
Comment (MackHalliday) [inline tests/app/v2/notifications/test_push_notifications.py:271]: Uses default app instead.

File: pyproject.toml [hunk 7]
```diff
@@ -6,6 +6,7 @@ authors = [
     "VA Notify <vanotify@va.gov>",
 ]
 readme = "README.md"
+package-mode = false
```
[the remark this answers is not in the document]
Comment (kalbfled) [reply pyproject.toml:9]: This is already in the ENP repo.  See [here](https://python-poetry.org/docs/basic-usage/#operating-modes).

[the remark this answers is not in the document]
Comment (EvanParish) [reply pyproject.toml:9]: nice. Thanks for adding this!

**Review by mchlwellman — DISMISSED** [review]

File: app/status/healthcheck.py [hunk 8]
```diff
@@ -1,10 +1,9 @@
-from flask import jsonify, Blueprint, request
+from flask import Blueprint, current_app, jsonify, request
```
[the remark this answers is not in the document]
Comment (MackHalliday) [reply app/status/healthcheck.py]: Updated - Although I am not sure how this happened either. `Pre-commit` is running.
Screenshot 2024-10-31 at 10 07 19 AM

**Review by mchlwellman — APPROVED** [review]

**Pull request merged**